web/libs/editor/src/tags/object/RichText/domManager.ts

```diff
@@ -720,14 +788,21 @@ export default class DomManager {
   }
 
   relativeOffsetsToGlobalOffsets(start: string, startOffset: number, end: string, endOffset: number) {
-    const startEl = this.domData.findElementByPath(start);
-    const endEl = this.domData.findElementByPath(end);
+    let startEl = this.domData.findElementByPath(start);
+    let endEl = this.domData.findElementByPath(end);
 
     if (!startEl || !endEl) {
       return undefined;
     }
+    if (!(startEl instanceof DDDynamicBlock)) {
+      startEl = this.domData.findTextBlock(startEl.start, "end") as DDDynamicBlock;
+    }
+    if (!(endEl instanceof DDDynamicBlock)) {
+      // It really should be "end" and not "start" as we are looking for the exact container by the start position
```

Comment:
this and next comments makes it even more confusing. what `startEl` and `endEl` are and why do we use `"end"` for start? `"end"` for `endEl` seems more obvious. Also why do you convert text block to dynamic block? Is it possible to define interfaces and use mostly them? like if any element in `DomData.elements` could have the same interface that would be awesome.